### ISSUE TITLE
[armresourcegraph] Error while un-marshilling response

### ISSUE THREAD

**Issue by vijayrajah** [Azure/azure-sdk-for-go#16715]
### Bug Report

Module: github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resourcegraph/armresourcegraph
version: 0.1.0 (latest -- as of 27-dec-2021)

The query response [here](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resourcegraph/armresourcegraph#QueryResponse) is defined as
```
// REQUIRED; Query output in JObject array or Table format.
	Data map[string]interface{} `json:"data,omitempty"`
```

instead it should be (a slice) or it should be a `interface {}` as defined in [SDK1](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go@v60.3.0+incompatible/services/resourcegraph/mgmt/2021-03-01/resourcegraph#QueryResponse)
```
// REQUIRED; Query output in JObject array or Table format.
	Data []map[string]interface{} `json:"data,omitempty"`
```

This leads to un-marshilling error while parsing the response by SDK.

`"unmarshalling type *armresourcegraph.QueryResponse: json: cannot unmarshal array into Go value of type map[string]interface {}"`

the JSON response is valid & `data` field in JSON response is an JSON ARRAY

**Comment by lirenhe** [comment]
@vijayrajah, this seems to be a Swagger issue. 
If the value for Data is an array of Json object, the definition should be 

    "Data": {
      "type": "array",
      "items": {
        "type": "object",
        "description": ""
      }

And the current definition is not correct.  https://github.com/Azure/azure-rest-api-specs/blob/7b89a1a3fcb08b6f9f12ee67049b6b8d561ad684/specification/resourcegraph/resource-manager/Microsoft.ResourceGraph/stable/2021-03-01/resourcegraph.json#L311

**Comment by tadelesh** [comment]
@vijayrajah We've released the 0.3.0 version of `armresourcegraph` which has resolved this issue. Could you please upgrade your dependency? Thanks.
